### ISSUE TITLE
Inverted cover option

### ISSUE THREAD

**Issue by noembryo** [koreader/koreader#12274]
…And another small feature that nobody really needs but...
Can we get another option in the Sleep screen>Wallpaper menu to show the book cover inverted?
Some times the Cover of a book is really dark and it doesn't look good on the reader.
Perhaps an inverted version will look better on e-ink?
Cosmetics.. 🤩

**Comment by Commodore64user** [comment]
legitimate question: if you really care, why don't you use calibre to get everything perfect (metadata, covers, etc) before you transfer the books to your e-reader?

I mean, you won't catch me with less than stellar covers on my readers.

**Comment by noembryo** [comment]
> if you really care, why don't you use calibre to get everything perfect (metadata, covers, etc) before you transfer the books to your e-reader?

What makes you think that I don't?

The book cover I use, is the one from the book itself.
Most of the time the cover is colorful and nice if you view it on a color monitor/device.
Nothing wrong with the cover per se.
The problem arise when you view it with e-ink's 16 shades of gray.
Sometimes the image looks really confusing if it's too dark.
Perhaps it will look better if most of the image is light colored (hence the feature).
Perhaps not.

As i said, a really minor enhancement.. 🤷‍♂️

**Comment by offset-torque** [comment]
**In case someone decides to implement this:**

I don't think the menu approach is the best one. This means user has to go into a long menu path and change this option every time they switch books with light/dark covers. Otherwise nice looking light cover will turn into  a dark mess.

A better approach might be (if implementable):
- Give a timeout of like 5 seconds when user turns off the device
- Show the cover normally but if the user taps on the screen during these 5 seconds, invert the colors.

There might be a message during this timeout **"Tap the screen to invert cover colors"** (or not if this causes ghosting etc.)

Advantage is no menu needs to be added and this is much quicker for the user. When they turn off their device they will see **"Oh I am reading this (dark cover) book"** and tap to invert. No option to change and other users won't be affected.

**Comment by NiLuJe** [comment]
First: ugh.

Second: inverted means inverted, it will *never* look good.

What you want is to fix the levels instead and that's already a manual "science" that is not as easy as you'd think to automate (e.g., something like IM's -normalize).

TL;DR: Triple no, and out of scope to boot, as we're not a signal processing suite ;).

**Comment by Commodore64user** [comment]
> I don't think the menu approach is the best one.

I agree with this, there are already at least 3 too many options to choose from in regard to the sleep screen (this is just my opinion, i won't say which. I am happy I was able to reduce the number by one when that was reworked).

> Give a timeout of like 5 seconds when user turns off the device 

On the other hand, this (according to me) is not a better approach either. I would hate to have to deal with this. Every time you lock the screen you have to wait 5 seconds to put your device in your bag avoiding at all costs that nothing touches it, nope thanks.

In all fairness though, i just don't think this is a good idea at all (no disrespect). There is a reason why images are not inverted when you are in night mode, they look really bad.

Also, until this ticket, i hadn't known this to be an issue at all, and I read quite a large number of Penguin Classic Editions which tend to have  covers on the darker side.

I really think that to solve "this problem", one just needs to make sure not to have too dark covers on their ebooks to begin with and all is good. 😊 and calibre will easily help you do that.

**Comment by noembryo** [comment]
So, it seems that it wasn't such a good idea after all..
Thanks for the consideration anyway.. 🙏

**Comment by offset-torque** [comment]
I agree with **NiLuJe** and **Commodore64user** . I started making my own screensaver images after seeing how awful many covers look since they are not designed with e-ink in mind. Even when I am starting with a suitable image, adjusting levels and choosing a suitable dithering algorithm to make it look nice takes some trial and error. I think there is a very low probability that a simple inversion will make a book cover look nicer.

**Comment by mergen3107** [comment]
My best suggestion would be:
- by hand, convert book cover to 16 gray scale
- Adjust white/black point to make them more contrasty
- Upload to KOReader 

I did this processing for NASA wallpapers once, they look great on eink! Especially the lighter brightness ones:
https://www.reddit.com/r/Onyx_Boox/comments/w9zili/nasa_posters_adapted_for_eink/

**Comment by mergen3107** [comment]
As a bonus, you can just extract covers, process them, then use the neat KOReader feature to set a custom cover :D

**Comment by mergen3107** [comment]
(If only `magick` was available in KOReader itself, this could be done on the fly with one slider… dreams, dreams :D)

**Comment by noembryo** [comment]
@mergen3107 I do edit cover with my usual workflow.
My test case for the feature request is "holidays".
No Calibre there and no PC.
Downloading a book with my mobile, send it to the reader, the cover is bad..
I was looking for an easy, fast improvement.

**Comment by Frenzie** [comment]
To offer a bit of a counter, I think inverted images can look quite cool.

That being said, the "perhaps" in the OP is slightly suspicious. ;-)

**Comment by noembryo** [comment]
> That being said, the "perhaps" in the OP is slightly suspicious. ;-)

Not a great sign of confidence is it? 😉

**Comment by offset-torque** [comment]
> I did this processing for NASA wallpapers once, they look great on eink!

Super! I was thinking of converting them and here I found them ready :) Thanks **mergen3107**

**Comment by NiLuJe** [comment]
> (If only `magick` was available in KOReader itself, this could be done on the fly with one slider… dreams, dreams :D)

It's shipped w/ the screensavers hack (which means it's in KoboStuff, too). be prepared to wait forever for even the most basic things, even on a Q8 build with "please don't kill the RAM" build settings.

**Comment by mergen3107** [comment]
@NiLuJe 
Even MTK would struggle? New Kobos, Kindle Scribe/PW5/new Basic

**Comment by NiLuJe** [comment]
It would be moderately less worse, but I wouldn't jump for joy ;p.

**Comment by Commodore64user** [comment]
> > That being said, the "perhaps" in the OP is slightly suspicious. ;-)
> 
> 
> 
> Not a great sign of confidence is it? 😉 

It sounds like you are expecting this to be a magical solution, yet you haven't even tried it. Otherwise you would have noticed 1) how bad things can get and 2) how not-so-simple a process it is.

**Comment by Commodore64user** [comment]
> It would be moderately less worse, but I wouldn't jump up for joy ;p.

I remember trying to use Photoshop with 4GB ram a while back, nothing fancy just a few simple adjustment layers. Nightmarish is how i would describe it. Yes, yes i know it isn't a fair comparison, point is retouching images takes a considerable amount of resources and eink is perhaps the last place i would consider for doing just that.

**Comment by Commodore64user** [comment]
@Frenzie here is a crazy idea. You know how every release has a nice illustration to go along with it. Why not make an e ink version that is shipped with every release. It could be used as the backup sleep screen image instead of that squared KOReader png that looks awful when one has images stretched.

**Comment by noembryo** [comment]
> It sounds like you are expecting this to be a magical solution, yet you haven't even tried it. Otherwise you would have noticed 1) how bad things can get and 2) how not-so-simple a process it is.

Really now?

**Comment by mergen3107** [comment]
I had an idea in the back of my mind before I honed the prost-processing of my mangas (which is now in the suer guide thanks to @offset-torque). 

Have this processing set up on a home server and have KOReader only download the end result. 

For manga that was too tough, my upscaled and dithered manga volumes sometimes reach 200 MB each. But for this cover it is quite doable :D 

Have KOReader upload the cover, home server converts and KOReader downloads it back :D
